honeybadger-client/src/main/java/com/workable/honeybadger/HoneybadgerClient.java

```diff
@@ -204,22 +217,30 @@ protected void doDispatchError(Error error) {
      * @param error
      * @return
      */
-    private boolean shouldExclude(Throwable error){
+    protected boolean shouldExclude(Throwable error){
 
         if (error == null){
             return false;
         }
 
         String errorClassName = error.getClass().getName();
-
-        if (errorClassName != null){
+        if (!isNullOrEmpty(errorClassName)){
             for (String className : excludedExceptionClasses){
                 if (errorClassName.startsWith(className)){
                     return true;
                 }
             }
         }
 
+        StackTraceElement[] stackTraceElements = error.getStackTrace();
```

Comment:
I would add a condition to check that excludedExceptionCauses is not empty to avoid the call to getStackTrace when no exceptionCauses are defined

Reply:
Done